src/audio/tdfb/tdfb.c

```diff
@@ -147,10 +176,109 @@ static int tdfb_init_coef(struct tdfb_comp_data *cd, int source_nch,
 		return -EINVAL;
 	}
 
-	coefp = ASSUME_ALIGNED(&config->data[0], 2);
+	if (config->num_angles > SOF_TDFB_MAX_ANGLES) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid num_angles %d",
+			    config->num_angles);
+		return -EINVAL;
+	}
+
+	if (config->beam_off_defined > 1) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid beam_off_defined %d",
+			    config->beam_off_defined);
+		return -EINVAL;
+	}
+
+	if (config->num_mic_locations > SOF_TDFB_MAX_MICROPHONES) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid num_mic_locations %d",
+			    config->num_mic_locations);
+		return -EINVAL;
+	}
+
+	if (config->num_angles == 0 && config->num_mic_locations == 0) {
```

Comment:
Can we have a comment here on why we set legacy ?

Reply:
Yep, I added above some comment text.